test/config-luatest/roles_test.lua

```diff
@@ -353,3 +353,164 @@ g.test_role_reload_error = function(g)
         exp_err = 'Error stopping role one: Wrongly stopped'
     })
 end
+
+-- Make sure dependencies in roles works as intended.
+g.test_role_dependencies_success = function(g)
+    local one = [[
+        local function apply()
+            _G.foo = _G.foo .. '_one'
+        end
+
+        return {
+            validate = function() end,
+            apply = apply,
+            stop = function() end,
+        }
+    ]]
+
+    local two = [[
+        local function apply()
+            _G.foo = _G.foo .. '_two'
+        end
+
+        return {
+            dependencies = {'one'},
+            validate = function() end,
+            apply = apply,
+            stop = function() end,
+        }
+    ]]
+
+    local three = [[
+        local function apply()
+            _G.foo = _G.foo .. '_three'
+        end
+
+        return {
+            dependencies = {'four', 'two'},
+            validate = function() end,
+            apply = apply,
+            stop = function() end,
+        }
+    ]]
+
+    local four = [[
+        _G.foo = 'four'
+
+        return {
+            validate = function() end,
+            apply = function() end,
+            stop = function() end,
+        }
+    ]]
+
+    local five = [[
+        local function apply()
+            _G.foo = _G.foo .. '_five'
+        end
+
+        return {
+            validate = function() end,
+            apply = apply,
+            stop = function() end,
+        }
+    ]]
+
+    local verify = function()
+        t.assert_equals(_G.foo, 'four_one_two_three_five')
+    end
+
+    helpers.success_case(g, {
+        roles = {one = one, two = two, three = three, four = four, five = five},
+        options = {
+            ['roles'] = {'four', 'three', 'two', 'one', 'five'}
+        },
+        verify = verify,
+    })
+end
+
+g.test_role_dependencies_error_wrong_type = function(g)
+    local one = [[
+        return {
+            dependencies = 'two',
+            validate = function() end,
+            apply = function() end,
+            stop = function() end,
+        }
+    ]]
+
+    helpers.failure_case(g, {
+        roles = {one = one},
+        options = {
+            ['roles'] = {'one'}
+        },
+        exp_err = 'Wrong type of role one dependencies'
```

Comment:
Nit: I propose to try to make the error more clear.

> Role "one" has field "dependencies" of type string, array-like table or nil expected

Reply:
Done.